File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/KeysMessage.java
package org.whispersystems.signalservice.api.messages.multidevice;


import org.whispersystems.signalservice.api.kbs.MasterKey;
import org.whispersystems.signalservice.api.storage.StorageKey;

import java.util.Optional;

public class KeysMessage {

  private final Optional<StorageKey> storageService;
  private final Optional<MasterKey>  master;

  public KeysMessage(Optional<StorageKey> storageService, Optional<MasterKey> master) {
    this.storageService = storageService;
    this.master         = master;
  }

  public Optional<StorageKey> getStorageService() {
    return storageService;
  }

  public Optional<MasterKey> getMaster() {
    return master;
  }
}

File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/KeysMessage.kt
package org.whispersystems.signalservice.api.messages.multidevice

import org.signal.core.models.AccountEntropyPool
import org.signal.core.models.MasterKey
import org.signal.core.models.backup.MediaRootBackupKey
import org.signal.core.models.storageservice.StorageKey

data class KeysMessage(
  val storageService: StorageKey?,
  val master: MasterKey?,
  val accountEntropyPool: AccountEntropyPool?,
  val mediaRootBackupKey: MediaRootBackupKey?
)
